lighthouse-cli/performance-experiment/server.js

```diff
@@ -21,96 +21,142 @@
-let lhParams;
 function hostExperiment(params, results) {
-  lhResults = results;
-  lhParams = params;
-
   return new Promise(resolve => {
+    database = new ExperimentDatabase(params.url, params.config);
+    const id = database.saveData(params.flags, results);
+    fallbackReportId = id;
+
     const server = http.createServer(requestHandler);
     server.listen(0);
-    server.on('listening', () => {
-      opn(`http://localhost:${server.address().port}/`);
-    });
+    server.on('listening', () => opn(`http://localhost:${server.address().port}/?id=${id}`));
     server.on('error', err => log.error('PerformanceXServer', err.code, err));
     server.on('close', resolve);
     process.on('SIGINT', () => {
+      database.clear();
       server.close();
     });
   });
 }
 
 function requestHandler(request, response) {
-  const pathname = parse(request.url).pathname;
-
-  if (request.method === 'GET') {
-    if (pathname === '/') {
-      reportRequestHandler(request, response);
-    } else if (pathname === '/blocked-url-patterns') {
-      blockedUrlPatternsRequestHandler(request, response);
+  request.parsedUrl = parse(request.url, true);
+  const pathname = request.parsedUrl.pathname;
+  try {
+    if (request.method === 'GET') {
+      if (pathname === '/') {
+        reportRequestHandler(request, response);
+      } else if (pathname === '/flags') {
+        flagsRequestHandler(request, response);
+      } else {
+        throw new HTTPError(404);
+      }
+    } else if (request.method === 'POST') {
+      if (pathname === '/rerun') {
+        rerunRequestHandler(request, response);
+      } else {
+        throw new HTTPError(404);
+      }
     } else {
-      response.writeHead(404);
-      response.end('404: Resource Not Found');
+      throw new HTTPError(405);
     }
-  } else if (request.method === 'POST') {
-    if (pathname === '/rerun') {
-      rerunRequestHandler(request, response);
+  } catch (err) {
+    if (err instanceof HTTPError) {
+      response.writeHead(err.statusCode);
+      response.end(err.message || http.STATUS_CODES[err.statusCode]);
     } else {
-      response.writeHead(404);
-      response.end('404: Resource Not Found');
+      response.writeHead(500);
+      response.end(http.STATUS_CODES[500]);
+      log.err('PerformanceXServer', err.code, err);
     }
-  } else {
-    response.writeHead(405);
-    response.end('405: Method Not Supported');
   }
 }
 
 function reportRequestHandler(request, response) {
-  const html = new PerfXReportGenerator().generateHTML(lhResults, 'perf-x');
-  response.writeHead(200, {'Content-Type': 'text/html'});
-  response.end(html);
+  try {
+    const id = request.parsedUrl.query.id || fallbackReportId;
+
+    const reportsMetadata = Object.keys(database.timeStamps).map(key => {
+      const generatedTime = database.timeStamps[key];
+      return {url: database.url, reportHref: `/?id=${key}`, generatedTime};
+    });
+    reportsMetadata.sort((metadata1, metadata2) => {
```

Comment:
something like this should work:
```
reportsMetadata.sort((a, b) => b.generatedTime - a.generatedTime); 
```

(maybe flip the two if the order is wrong ;)